parlai/agents/rag/rag.py

```diff
@@ -402,13 +403,48 @@ def update_state_dict(
             )
         return state_dict
 
+    def _should_override_dpr_model_weights(self, opt: Opt):
+        """
+        Determine if we need to override the DPR Model weights.
+
+        Under certain circumstances, one may wish to specify a different
+        `--dpr-model-file` for a pre-trained, RAG model. Thus, we additionally
+        check to make sure that the loaded DPR model weights are not overwritten
+        by the state loading.
+
+        NOTE: If `--model-file M` was trained with `--dpr-model-file D`, and
+        `--dpr-model-file D` is specified *after training* (i.e., in eval/interactive),
+        we cannot currently handle replacing those weights.
+        """
+        try:
+            init_model, _ = self._get_init_model(opt, None)
+            init_model_opt = Opt.load(f'{init_model}.opt')
+            override_dpr = modelzoo_path(
+                opt['datapath'], opt['dpr_model_file']
+            ) != modelzoo_path(opt['datapath'], init_model_opt['dpr_model_file'])
+            if override_dpr:
+                logging.warning(
+                    f"Overriding DPR Model with {modelzoo_path(opt['datapath'], opt['dpr_model_file'])}"
+                )
+        except FileNotFoundError:
```

Comment:
should there be warning here?

Reply:
hmm no, since it's supposed to be silent to the user